server/src/main/java/thesistrack/ls1/service/ThesisService.java

```diff
@@ -324,6 +360,17 @@ public Thesis findById(UUID thesisId) {
                 .orElseThrow(() -> new ResourceNotFoundException(String.format("Thesis with id %s not found.", thesisId)));
     }
 
+    public ThesisPresentation findPresentationById(UUID thesisId, UUID presentationId) {
```

Comment:
The naming "find presentation **by ID** suggest that only the `presentationId` is required as a parameter. Why do you need the `thesisId` here?

Reply:
Its not really required to have it but it prevents security issues because otherwise it could be possible to request presentations of a thesis you dont have access to (if you dont add the check in the calling code).
I prefer to have this check in this method to not forget it when calling it